examples/pipelines/datacomp/components/dedup_image_embeddings/src/main.py

```diff
@@ -0,0 +1,89 @@
+"""This component deduplicates images based on embeddings.
+
+As proposed in [Abbas et al., 2022](https://arxiv.org/abs/2303.09540).
+
+Implementation is based on https://github.com/conceptofmind/SemDeDup/tree/main.
+"""
+import logging
+
+import dask.dataframe as dd
+import numpy as np
+import pandas as pd
+from scipy.spatial.distance import cdist
+from sklearn.metrics.pairwise import cosine_similarity
```

Comment:
Maybe we can pass the similarity function as argument to the component? I think for some use cases it could be useful to use different similarity methods as well.

Reply:
Which other similarities are you thinking of? Cause they might require specific sklearn functions

Reply:
For my use case I need the jaccard similarity. I modified the component slightly. Maybe we can try to combine it later on. For now, I will use my custom one.